lib/semantics/check-purity.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2018-2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "check-purity.h"
+#include "tools.h"
+#include "../parser/parse-tree.h"
+
+namespace Fortran::semantics {
+void PurityChecker::Enter(const parser::ExecutableConstruct &exec) {
+  if (InPureSubprogram() && IsImageControlStmt(exec)) {
+    context_.messages().Say(GetImageControlStmtLocation(exec),
```

Comment:
I think you can replace `context_.messages().Say` with `context_.Say`.

Reply:
Ok.